fontpreview

```diff
@@ -155,18 +148,28 @@ trap "" SIGTSTP
 
 trap pre_exit EXIT
 
+# Use mktemp to create a temporary directory that won't
+# collide with temporary files of other application.
+FONTPREVIEW_DIR="$(mktemp -d "${TMPDIR:-/tmp}/fontpreview_dir.XXXXXXXX")" || exit
+PIDFILE="$FONTPREVIEW_DIR/fontpreview.pid"
+touch "$PIDFILE" || exit
+FONT_PREVIEW="$FONTPREVIEW_DIR/fontpreview.png"
+touch "$FONT_PREVIEW" || exit
+TERMWIN_IDFILE="$FONTPREVIEW_DIR/fontpreview.termpid"
+touch "$TERMWIN_IDFILE" || exit
```

Comment:
Could you explain why you are using `exit` after commands for creating the files? 

And why we need to create the files in the start? Cant it be made and written? In the case of the `FONT_PREVIEW`, we create the file using `convert` later in the script, so what is the use for creating the file beforehand? Could this be a OpenBSD thing?

Reply:
> Could you explain why you are using `exit` after commands for creating the files?

The idea here is to abort the script if something goes wrong.

> And why we need to create the files in the start? 

This is how it is done at the moment. I didn't want to rewrite everything from scratch so I sticked the existing code.

>Cant it be made and written? In the case of the `FONT_PREVIEW`, we create the file using `convert` later in the script, so what is the use for creating the file beforehand? 

Absolutely, if there's no need to touch those files in the beginning in the current version of the program then this could be patched out so that we just create some of the files as we go.

> Could this be a OpenBSD thing?

I am using FreeBSD, but I am pretty sure that there's no difference in behavior in this case between any general purpose Unix-like systems.

Reply:
> This is how it is done at the moment. I didn't want to rewrite everything from scratch so I sticked the existing code.

Oh sorry, my bad. I did not see that this was already in the code 😅